### PR TITLE
test: Start of porting InfluxRpc query_tests

### PR DESCRIPTION
**Pull request by carols10cents**
Connects to #6617.

I've ported two tests from [`query_tests/influxrpc/field_columns`](https://github.com/influxdata/influxdb_iox/blob/6f39ae342e64848bd6555bddbc1d3fa30050f75e/query_tests/src/influxrpc/field_columns.rs#L51-L86) and made them more in the style of the tests already existing in [`influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata.rs`](https://github.com/influxdata/influxdb_iox/blob/6f39ae342e64848bd6555bddbc1d3fa30050f75e/influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata.rs).

I think I have an abstraction that should work for the other kinds of InfluxRpc tests, namely:

- A new trait, `InfluxRpcTest`, that types can implement to define how to run a test on a specific Storage gRPC API. `InfluxRpcTest` takes care of iterating through the two architectures, running the setups, and creating the custom test step.
- Implementers of the trait like `MeasurementFieldsTest` in this PR can define aspects of the tests that differ per run, to make the parameters of the test clearer and highlight what different tests are testing.

### CODE REVIEW

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata.rs [hunk 1]
```diff
@@ -281,3 +284,80 @@ async fn measurement_fields() {
     )
     .await
 }
+
+#[tokio::test]
+async fn field_columns_nonexistent_table_with_predicate() {
```
Comment (alamb) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/metadata.rs:289]: 👌 

Looks very nice to me

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc.rs [hunk 2]
```diff
@@ -56,3 +60,80 @@ pub(crate) fn read_group_data() -> Vec<&'static str> {
         "cpu,cpu=cpu2,host=bar  usage_user=52.0,usage_system=41.0 2000",
     ]
 }
+
```
Comment (alamb) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc.rs:63]: FWIW the reason to have end to end rpc tests was to test the end to end plumbing when query_tests were some simulation

Now that the new query_tests are running as end to end tests, we can probably consolidate them all

**Review by alamb — APPROVED** [review]
The framework looks great to me @carols10cents  -- thank you ❤️ 

CI appears to be failing but otherwise this PR looks good to me

**Comment by carols10cents** [comment]
```
Server command failed: Error in ingester subcommand: Catalog DSN error: A catalog error occurred: 
unhandled sqlx error: pool timed out while waiting for an open connection
```

😬 Sooo the persist API is just persisting everything, and when tests run in parallel they interfere with each other when they call the persist API, so these tests all start non-shared `MiniCluster`s, and it looks like all the postgres connections are getting exhausted :( 

Let me see if I can change the persist API for both architectures to persist for a particular namespace instead.

**Comment by carols10cents** [comment]
mmmmm i think i'm making different race conditions now 😅 Will pick this up on monday.

**Pull request merged**